pennylane/templates/state_preparations/mottonen.py

```diff
@@ -268,36 +229,66 @@ def MottonenStatePreparation(state_vector, wires):
 
         The final state is only equal to the input state vector up to a global phase.
 
+    .. warning::
+
+        Due to non-trivial classical processing of the state vector,
+        this template is not always fully differentiable.
+
     Args:
-        state_vector (tensor_like): Input array of shape ``(2^N,)``, where N is the number of wires
-            the state preparation acts on. The input array must be normalized, and ``N`` must be smaller
-            or equal to the total number of wires.
-        wires (Iterable or Wires): Wires that the template acts on. Accepts an iterable of numbers or strings, or
-            a Wires object.
-
-    Raises:
-        ValueError: if inputs do not have the correct format
+        state_vector (tensor_like): Input array of shape ``(2^n,)``, where ``n`` is the number of wires
+            the state preparation acts on. The input array must be normalized.
+        wires (Iterable): wires that the template acts on
+
     """
 
-    ###############
-    # Input checks
+    num_params = 1
+    num_wires = AnyWires
+    par_domain = "A"
+
+    def __init__(self, state_vector, wires, do_queue=True):
+
+        shape = qml.math.shape(state_vector)
+
+        if len(shape) != 1:
+            raise ValueError(f"State vector must be a one-dimensional vector; got shape {shape}.")
+
+        n_amplitudes = shape[0]
+        if n_amplitudes != 2 ** len(wires):
+            raise ValueError(
+                f"State vector must be of length {2 ** len(wires)} or less; got length {n_amplitudes}."
+            )
+
+        norm = qml.math.sum(qml.math.abs(state_vector) ** 2)
+        if not qml.math.allclose(norm, 1.0, atol=1e-3):
+            raise ValueError("State vector has to be of length 1.0, got {}".format(norm))
```

Comment:
```suggestion
            raise ValueError("State vector has to be of norm 1.0, got {}".format(norm))
```

Two different types of errors are "State vector must be of length ...". The two different uses of the word length are confusing.  Wrong size of `state_vector` should use a different word than wrong normalization of `state_vector`.

Reply:
Good point!